src/components/InputComponent/InputComponent.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from 'react';
+import { IInputProps } from './inputComponent.types';
+import './styles.scss';
+
+export const InputComponent: React.FC<IInputProps> = ({
+  border,
+  dark,
+  disabled,
+  error,
+  errorMessage,
+  inputValueReciever,
+  ...rest
+}) => {
+  const [value, setValue] = useState<string>('');
+
+  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+  };
+
+  useEffect(() => {
+    if (inputValueReciever) {
+      inputValueReciever(value);
+    }
+  }, [value, inputValueReciever]);
+
+  return (
+    <div className='input-wrapper'>
+      <input
+        value={value}
+        onChange={inputChangeHandler}
```

Comment:
please remove `inputChangeHandler` and pass `onChange` as a prop to the component to make it reusable.
 
 Also lift state up for `value` as well, passing it as a prop

Reply:
https://github.com/igorizviekov/mt-booking_app/pull/24/commits/74ff661575e673759d0bc1baf2f28de96e5ddb28